ffi/src/winscard/system_scard/card.rs

```diff
@@ -0,0 +1,403 @@
+use std::borrow::Cow;
+use std::mem::size_of;
+use std::ptr::null_mut;
+use std::slice::from_raw_parts;
+
+#[cfg(target_os = "windows")]
+use ffi_types::winscard::functions::SCardApiFunctionTable;
+use ffi_types::winscard::ScardIoRequest;
+#[cfg(target_os = "windows")]
+use ffi_types::winscard::{ScardContext, ScardHandle};
+use num_traits::ToPrimitive;
+use winscard::winscard::{
+    AttributeId, ControlCode, IoRequest, Protocol, ReaderAction, ShareMode, Status, TransmitOutData, WinScard,
+};
+use winscard::{Error, ErrorKind, WinScardResult};
+
+use super::parse_multi_string_owned;
+use crate::winscard::buf_alloc::SCARD_AUTOALLOCATE;
+#[cfg(not(target_os = "windows"))]
+use crate::winscard::pcsc_lite::functions::PcscLiteApiFunctionTable;
+#[cfg(not(target_os = "windows"))]
+use crate::winscard::pcsc_lite::{initialize_pcsc_lite_api, ScardContext, ScardHandle};
+
+/// Represents a state of the current `SystemScard`.
+#[derive(Copy, Clone)]
+enum HandleState {
+    /// The card is not connected or has been disconnected.
+    Disconnected,
+    /// The card is connected and ready to use.
+    Connected(ScardHandle),
+}
+
+pub struct SystemScard {
+    h_card: HandleState,
+    h_card_context: ScardContext,
```

Comment:
question: What does the `h_` prefix stands for?

Reply:
`h_` stands for _handle_. I wanted to mark somehow that the value is just a handle to the corresponding card or card context. I saw a similar naming in the MSDN. For example, the [SCardConnectW](https://learn.microsoft.com/en-us/windows/win32/api/winscard/nf-winscard-scardconnectw) function.
Also, we already have a lot of things called `scard` and `scard_context`. It can confuse the user why we have the scard inside another scard, and so on.